lib/util/defineRuleString.js

```diff
@@ -0,0 +1,24 @@
+const {definePatternRule} = require('math-rules');
+const {parse} = require('math-parser');
+const {traverse} = require('math-traverse');
+
+const defineRuleString = (matchPattern, rewritePattern, constraints) => {
+  const matchAST = parse(matchPattern);
+  const rewriteAST = parse(rewritePattern);
+
+  traverse(matchAST, {
```

Comment:
There parser used add `loc` props on each node describing substring range within the original source string that maps to the node.  The parser no longer does that.  I forgot to update this code.  I'll remove the code that delete `loc` as it's no longer necessary.